engine/runtime-compiler/src/main/java/org/enso/compiler/context/NameResolution.java

```diff
@@ -0,0 +1,61 @@
+package org.enso.compiler.context;
+
+import java.util.Optional;
+import org.enso.compiler.MetadataInteropHelpers;
+import org.enso.compiler.core.ConstantsNames;
+import org.enso.compiler.core.ir.Name;
+import org.enso.compiler.data.BindingsMap;
+import org.enso.compiler.pass.analyse.AliasAnalysis$;
+import org.enso.compiler.pass.analyse.alias.Info;
+import org.enso.compiler.pass.resolve.GlobalNames$;
+import scala.Option;
+
+/**
+ * Encapsulates the logic for resolving a variable name.
+ *
+ * <p>The same logic is needed in two places: 1. in the runtime ({@link
+ * org.enso.interpreter.runtime.IrToTruffle.processName}), 2. in the type checker ({@link
+ * org.enso.compiler.pass.analyse.types.TypeInference.processName}).
+ *
+ * <p>To ensure that all usages stay consistent, they should all rely on the logic implemented in
+ * this class, customizing it to the specific needs of the context in which it is used.
+ *
+ * @param <ResultType> The result type of the resolution process.
+ * @param <LocalNameLinkType> The type describing a link to a local name in a current scope.
+ *     Depending on the context this may be a Graph.Link (in the compiler) or a FramePointer (in the
+ *     runtime).
+ */
+public abstract class NameResolution<ResultType, LocalNameLinkType> {
```

Comment:
I suggest to add `Algorithm` suffix to these classes, just like [ImportResolverAlgorithm](https://github.com/enso-org/enso/pull/9539/files#r1542503066) does.

Reply:
Ok, thanks for the suggestion, will do.